lib/streams.js

```diff
@@ -57,13 +57,26 @@ class BlobWriteStream extends Writable {
   _final (cb) {
     this._append(err => {
       if (err) return cb(err)
+
       this.id.blockLength = this.core.length - this.id.blockOffset
       this.id.byteLength = this.core.byteLength - this.id.byteOffset
+
       this._release()
+      this._release = null
+
       return cb(null)
     })
   }
 
+  _destroy (cb) {
+    if (this._release) {
+      this._release()
+      this._release = null
+    }
+
+    cb(null)
+  }
```

Comment:
Unsure if this is the most precise fix but seems ok

Reply:
_destroy is always called, so just release it there always and remove the releasing in _final